### ISSUE TITLE
<windows.h> does not exist on linux

### ISSUE THREAD

**Issue by 506independent** [HackerPoet/FractalSoundExplorer#2]
windows.h does not exist on linux. i am unable to compile and use. could you just make this cross-platform? i know it would be a lot of work, but i really want to try your program. 

here is the output of my compiler:

```
$ g++ Main.cpp WinAudio.cpp
In file included from Main.cpp:3:
WinAudio.h:3:10: fatal error: windows.h: No such file or directory
    3 | #include <windows.h>
      |          ^~~~~~~~~~~
compilation terminated.
In file included from WinAudio.cpp:1:
WinAudio.h:3:10: fatal error: windows.h: No such file or directory
    3 | #include <windows.h>
      |          ^~~~~~~~~~~
compilation terminated.
```

stop your windows-only party, and let linux users in!

**Comment by AuracleTech** [comment]
When you make something you make it for the majority of users (98% rule).
Chairs shouldn't be 1 feet tall just because 0.006% of the population are dwarfs .
They adapt to us.

Same principle here. You use linux, you adapt.

Note that this doesn't mean this issue shouldn't be solved, just that your statement 

> stop your windows-only party, and let linux users in!

Is not relevant

**Comment by Tremeschin** [comment]
Don't think we should fight (yet again) for this fallacy of "majority of users".

I disagree with you two, rushing Linux support the way we're asking here is not how open source works, stuff needs time and effort to be implemented, so just click that fork button and write a Linux compatible code.

Also Linux is just way easier to develop with and reliable IMHO, 95% the issues with all the three projects I worked on Windows with had to do with some internal trickery and mainly file permissions or having the users compile or use compiled binaries. Also having to write long texts than simple commands for users to run is pretty annoying.

Don't get me wrong, I'm not being personal here nor aggressive towards any of you two, we all should be welcoming in FOSS community and give opinions on caveats and likes, dislikes. Just my point of view as a small hobby developer on these never ending fights.

Anyways @506independent I'm forking the project and will try cross compiling for Windows and running the project with wine, that's how we adapt now. I don't know enough C++ to implement a Linux native code but this cross compile should make it runnable here and I'm ultra curious on toying with the project.

**Comment by mjrochford** [comment]
He already used a cross platform graphics library he just used windows only sound code, I ported it to mess around with [here](https://github.com/mjrochford/FractalSoundExplorer)

**Comment by Tremeschin** [comment]
> He already used a cross platform graphics library he just used windows only sound code, I ported it to mess around with [here](https://github.com/mjrochford/FractalSoundExplorer)

Ah that's definitely better than cross compiling, thank you, I was very close getting it to work just some weirdo Windows headers missing but was a fun journey, learned something new. Didn't lurk forks before so that's half my bad :p

**Comment by 506independent** [comment]
> He already used a cross platform graphics library he just used windows only sound code, I ported it to mess around with [here](https://github.com/mjrochford/FractalSoundExplorer)

im just wondering about how i would compile? here is my attempt:

```
$ g++ Main.cpp WinAudio.cpp
Main.cpp:4:10: fatal error: SFML/Audio.hpp: No such file or directory
    4 | #include <SFML/Audio.hpp>
      |          ^~~~~~~~~~~~~~~~
compilation terminated.
In file included from WinAudio.cpp:2:
WinAudio.h:3:10: fatal error: windows.h: No such file or directory
    3 | #include <windows.h>
      |          ^~~~~~~~~~~
compilation terminated.
```

is it unfinished, or

**Comment by sausix** [comment]
nivker has forked this and added independend audio support by SFML library.

https://github.com/nivkner/FractalSoundExplorer

**Comment by Nv7-GitHub** [comment]
How do you compile this fork?

**Comment by nivkner** [comment]
`g++ Main.cpp  -lsfml-system -lsfml-window -lsfml-graphics -lsfml-audio -lGL -I/usr/include/SFML -o FractalSoundExplorer`

**Comment by NicolasGoeddel** [comment]
l also had to install `libsfml-dev` before I was able to compile. I am able to run the program now on Ubuntu 20.10 but I guess my hardware is too old. The whole system begins to freeze when trying to zoom in and when I click on a region to hear the sound there are just very bad audio buffer overruns. :laughing:

**Comment by NSGolova** [comment]
I have ported this project even further on WebGL. https://github.com/radulov/FractalSoundWeb 
It should work on Linux too =)

**Comment by gek169** [comment]
Yeah I told him this was an issue and he said "nah everybody uses windows"
dude's using sfml and windows-only headers...
I bet he compiles the project on visual studio too for that e x t r a slowdown

**Comment by techsy730** [comment]
I can't speak to any discussions while this was a "fresh" project.
But asking someone to completely port a small side project to a very different standard library for the sake of, let's be honest, a small percent of his userbase (and I say that as _one_ of those in that small percent) is rather unreasonable.

Yeah, platform "more independent" libraries as a starting point for newer projects would be appreciated, but the ship has sailed for this one.

And it is not like we don't have tools, like Wine (or rather winelib, which is wine's implementation of the windows API Linux binaries can link against). And of course this is open source; we can collaborate on such a porting effort.
